platforms/core-configuration/model-core/src/main/java/org/gradle/api/internal/provider/AbstractProperty.java

```diff
@@ -190,7 +188,7 @@ public ValueProducer getProducer() {
 
     @Override
     public void finalizeValue() {
-        if (state.shouldFinalize(getDisplayName(), producer)) {
+        if (state.shouldFinalize(this::getDisplayName, producer)) {
```

Comment:
❓ Since the `displayName` argument doesn't vary on each call-site, would it be worth passing it at construction?

Reply:
The supplier, you mean? Yeah, that would work too. The value itself is mutable (in `Property`, via `OwnerAware.attachOwner(...)`).

Reply:
Actually, that would mean `FinalizedValue` (which all properties - and in the future, CFCs - share as their finalized state) could no longer be a singleton. I think passing the display name as a parameter where needed is a reasonable   shortcoming, compared to forcing a private Value State peer instance on each property/CFC/... instance.

Reply:
I replaced the use of `Supplier<DisplayName>` with a `Describable`, which preserves laziness and allows for clients that don't keep a `DisplayName` around (such as `DefaultCFC`).